atom/common/node_bindings.cc

```diff
@@ -213,19 +215,62 @@ void NodeBindings::Initialize() {
   // Explicitly register electron's builtin modules.
   RegisterBuiltinModules();
 
-  // Init node.
-  // (we assume node::Init would not modify the parameters under embedded mode).
-  // NOTE: If you change this line, please ping @codebytere or @MarshallOfSound
-  int argc = 0;
+  // pass non-null program name to argv so it doesn't crash
+  // trying to index into a nullptr
+  int argc = 1;
   int exec_argc = 0;
-  const char** argv = nullptr;
+  const char* prog_name = "electron";
+  const char** argv = {&prog_name};
   const char** exec_argv = nullptr;
+
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  if (env->HasVar("NODE_OPTIONS")) {
+    std::vector<std::string> disallowed = {
+        "--inspect", "--inspect-brk",    "--inspect-port", "--require",
+        "-r",        "--openssl-config", "--loader",       "--use-openssl-ca"};
+
+    std::string options;
+    bool needs_overwrite = false;
+    env->GetVar("NODE_OPTIONS", &options);
+    std::vector<std::string> parts = base::SplitString(
+        options, " ", base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+
+    // parse passed options for unsupported options
+    // and remove them from the options list
+    for (std::string part : parts) {
+      for (std::string disallow : disallowed) {
+        std::size_t found = disallow.find(part);
+        if (found != std::string::npos) {
+          LOG(WARNING) << "This NODE_OPTION is not supported in Electron";
+          parts.erase(std::remove(parts.begin(), parts.end(), part),
```

Comment:
You _really _ don't want to be modifying the `parts` vector while iterating through it in the outermost loop here.

I'd suggest you do something more like

```
std::vector<std::string> tmp;
for (const auto& part : parts) {
  if (disallow.count(part)) {
    warning...
    continue
  }
  tmp.push_back(part)
}
std::swap(parts, tmp)
```